scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -0,0 +1,544 @@
+```
+
+The `Tumor_Myoblast` cell type is the most prominent cell type, and based on the UMAPs is found in all 6 of our samples.
+We will start by comparing the gene expression of the myoblasts across ARMS and ERMS samples.
+
+
+## Differential Expression Analysis
+
+### Pseudo-bulking 
+
+Before we can compare the gene expression profiles of myoblasts in ARMS vs. ERMS patients, we will need to "pseudo-bulk" the gene counts. 
+Pseudo-bulking creates a new counts matrix that contains the total counts across all cells of a given label (e.g. cell type) for each sample. 
+This allows us to use single-cell resolution to define the labels and sum gene counts across groups of cells containing the same label, and avoids counting each cell as its own replicate. 
+
+Pseudo-bulking is implemented prior to differential expression analysis for the following reasons: 
+
+- Produces larger counts, which are more amenable to standard normalization and differential expression methods used by bulk RNA-sequencing. 
+-  Collapses gene expression counts by sample, so that samples, rather than cells, represent replicates.
+- Masks variance within a sample to highlight variance across samples.
+
+We are staring with the simple comparison of looking at one cell type and comparing across two groups of samples, samples belonging to either the ERMS and ARMS subtypes. 
+When we do this, we will first want to pseudo-bulk our dataset to group our cells by cell type and by sample.
+This will create a new `SingleCellExperiment` object that contains the pseudo-bulked counts to be used as input for differential expression analysis. 
+Then we can subset the pseudo-bulked object to just contain our cell type of interest before proceeding. 
+
+Before we apply pseudo-bulking to our dataset, lets look at a simple example of how pseudo-bulking works. 
+
+```{r}
+# create an example counts matrix
+counts_mtx <- matrix(1:12, 
+                     ncol = 4,
+                     dimnames = list(c("geneA", "geneB", "geneC"),
+                                     c("A-cell1", "A-cell2", "B-cell1", "B-cell2")))
+counts_mtx
+```
+
+
+```{r}
+# create new matrix, summing the counts for each gene across all cells in the specified group
+pseudobulked_counts <- cbind(
+  rowSums(counts_mtx[,1:2]), # sum all counts for cells labeled with group A
+  rowSums(counts_mtx[,3:4]) # sum all counts for cells labeled with group B
+)
+pseudobulked_counts 
+```
+Now the actual pseudo-bulking for our dataset! 
+We can pseudo-bulk using any groups that we are interested in.
+For right now, we would are interested in looking at gene expression across cell types, so we want to group the counts matrix by both cell type and sample. 
+
+```{r}
+# first let's subset our coldata to only have the columns we care about in pseudobulking 
+pseudobulked_coldata <- colData(rms_sce)[, c("celltype_broad", "sample")]
+
+# this creates a new SCE object that contains the pseudobulked counts across the provided groups 
```

Comment:
```suggestion
# this creates a new SCE object that contains the pseudo-bulked counts across the provided groups 
```

Reply:
Small note, we'll want to really emphasize that the _counts_ get summed, not the logcounts (by default).